daemon/src/model.rs

```diff
@@ -1275,4 +1276,47 @@ mod tests {
 
         assert_eq!(balance, SignedAmount::from_sat(1000))
     }
+
+    #[test]
+    fn proportional_funding_fees_if_sign_of_funding_rate_changes() {
+        let price = Price::new(Decimal::from_u32(35_000).unwrap()).unwrap();
+        let contracts = Usd::new(Decimal::from_u32(100).unwrap());
+        let long_leverage = Leverage::new(2).unwrap();
+        let long_pays_short_fee = calculate_funding_fee(
+            price,
+            contracts,
+            long_leverage,
+            FundingRate::new(Decimal::from_f64(0.01).unwrap()).unwrap(),
+            SETTLEMENT_INTERVAL.whole_hours(),
+        )
+        .unwrap();
+
+        let short_pays_long_fee = calculate_funding_fee(
+            price,
+            contracts,
+            long_leverage,
+            FundingRate::new(Decimal::from_f64(-0.01).unwrap()).unwrap(),
+            SETTLEMENT_INTERVAL.whole_hours(),
+        )
+        .unwrap();
+
+        assert_eq!(
+            long_pays_short_fee.fee,
+            short_pays_long_fee.fee * long_leverage.get().into()
+        )
+    }
+
+    #[test]
+    fn zero_funding_fee_if_zero_funding_rate() {
+        let fee = calculate_funding_fee(
+            Price::new(Decimal::from_u32(35_000).unwrap()).unwrap(),
+            Usd::new(Decimal::from_u32(100).unwrap()),
+            Leverage::new(2).unwrap(),
+            FundingRate::new(Decimal::ZERO).unwrap(),
```

Comment:
Extract into variable and make it the "arrange" part to make it more obvious what the influential variable is in the test?